### ISSUE TITLE
Changes to symlinked includes arent detected by smart/lazy compilation process

### ISSUE THREAD

**Issue by trullock** [platformio/platformio-core#4040]
### Configuration

**Operating system**: Windows 10

**PlatformIO Version**: Core - 5.1.1

### Description of problem

Changes to symlinked files inside (at least) `/include` are not detected.
I haven't tested other project dirs

#### Steps to Reproduce

Have a symlink inside `/include` to a file elsewhere on disc.
Compile your code
All is well
Modify only the symlinked file
Recompile
The changes aren't detected and so the code isn't recompiled

### Actual Results

Changed symlinked file isnt detected for recompilation

### Expected Results

Changes are detected and recompiled

### If problems with PlatformIO Build System:

**The content of `platformio.ini`:**
```
[env:esp32]
platform = espressif32
board = esp32dev
board_build.partitions = no_ota.csv
framework = arduino
```

**Source file to reproduce issue:**
```main.cpp

// both of these suffer the problem
#include <symlinkedfile.h>
#include "../include/symlinkedfile.h"
```

**Comment by valeros** [comment]
Hi @trullock ! I put together a simple example where a special configuration header file was symlinked to a file in a different location:

![symlink](https://user-images.githubusercontent.com/886732/134489496-66051234-b6b7-4f91-a197-1b94bbab1165.png)

It's strange, but I couldn't reproduce the issue, PlatformIO detected all changes I made in the original file and recompiled my project successfully. Could you please try to upgrade your PlatformIO installation to the latest available 5.2.0? Simply run `pio upgrade` in your terminal.

**Comment by trullock** [comment]
Can't replicate with 5.2.0 👍
